### DISCUSSION TITLE
Storyshots: Why am I getting an error saying `then` is undefined?

### DISCUSSION THREAD

**Discussion by jwir3** [storybookjs/storybook#17079] [Help]
I have an existing storybook that I would like to use `addons-storyshots` with. However, when I follow the steps listed [here](https://storybook.js.org/addons/@storybook/addon-storyshots), I get the following error:

```
 FAIL  src/test/visual/Storyshots.test.js
  ● Test suite failed to run

    TypeError: Cannot read property 'then' of undefined

      1 | import initStoryshots from '@storybook/addon-storyshots';
      2 |
    > 3 | initStoryshots();
        | ^
      4 |

      at testStorySnapshots (node_modules/@storybook/addon-storyshots/dist/ts3.9/api/index.js:54:66)
      at Object.<anonymous> (src/test/visual/Storyshots.test.js:3:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        7.242 s
Ran all test suites.
```
I have added a zip file of the repository in the state I am trying to run [here](http://assets.loonslanding.beer/aegir.tar.gz). Note that I'm only trying to run StoryShots for the `StandardButton` component at the moment (just to get the snapshotting working before I extend to other, more complicated components).

My test setup is in `src/test/visual/Storyshots.test.js`:
```
import initStoryshots from '@storybook/addon-storyshots';

initStoryshots({
    storyNameRegex: /StandardButton/,
});
```

I am using React 16, and so I originally followed the instructions for React 16, but the error I'm getting doesn't actually look like the error presented on that page (It appears to be crashing before I get to that point). It looks similar to [this question](https://github.com/storybookjs/storybook/issues/10455), but I don't know how to verify that my setup is the same as that one, so I am having difficulty debugging it.

Can anyone give me some steps to troubleshoot this issue?

**Comment by songz** [comment]
same issue for me.

**Reply by songz** [reply]
Our solution is to upgrade **all storyshot addons** to the same version and it fixed our issue: https://github.com/garageScript/c0d3-app/pull/1235 good luck!

**Reply by eric-burel** [reply]
And the reverse is true: if you just installed the addon say with version 6.4, also upgrade Storybook to 6.4 or install an older version of the addon if you can't, so versions matches.
